cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -663,7 +669,38 @@ func newStateMachine(disk disks.Disk) *stateMachine {
 func (m *stateMachine) execute() (finished bool, err error) {
 	next, err := m.current()
 	m.current = next
-	return next == nil, err
+	finished = next == nil
+	if finished && err == nil {
+		if err := m.finalize(); err != nil {
+			return true, err
+		}
+	}
+	return finished, err
+}
+
+func (m *stateMachine) finalize() error {
+	// check soundness
+	// the grade check makes sure that if data was mounted unecrypted
+	// but the model is secured it will end up marked as untrusted
+	isEncrypted := m.isEncryptedDev || m.model.Grade() == asserts.ModelSecured
+	part := m.degradedState.partition("ubuntu-data")
```

Comment:
thanks for this, I was also wondering about what if a secured system ended up thinking it was an unencrypted system this way, but best I can tell if they did that they would just boot a totally different system and still not have access to any of the encrypted data, similar to if they just flashed their own OS on the disk.